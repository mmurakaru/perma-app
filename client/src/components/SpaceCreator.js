import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ReactComponent as Sunflower } from '../assets/sunflower.svg'

SpaceCreator.propTypes = {
    updateTitle: PropTypes.func.isRequired,
    submitDisabled: PropTypes.bool,
    submitHandler: PropTypes.func.isRequired,
}

export default function SpaceCreator({
    updateTitle,
    submitDisabled,
    submitHandler,
}) {
    return (
        <BackgroundStyled>
            <NewSpaceContainer>
                <SpaceImageWrapper>
                    <SunflowerIcon />
                </SpaceImageWrapper>
                <SpaceNameInput>
                    <input
                        onChange={(event) => updateTitle(event.target.value)}
                        id="spacetitle"
                        type="text"
                    />
                </SpaceNameInput>
                <label htmlFor="spacetitle">
                    What would you like to name your space?
                </label>
                <ButtonWrapper>
                    <ButtonStyled
                        onClick={submitHandler}
                        disabled={submitDisabled}
                    >
                        Agree and create space
                    </ButtonStyled>
                </ButtonWrapper>
            </NewSpaceContainer>
        </BackgroundStyled>
    )
}

const BackgroundStyled = styled.section`
    background: var(--light-grey);
    width: 100%;
    height: 100%;
    padding: 20px;
`

const SunflowerIcon = styled(Sunflower)`
    width: 60%;
`

const NewSpaceContainer = styled.form`
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 15vh;

    label {
        font-weight: 200;
        width: 70%;
        text-align: center;
    }
`

const SpaceImageWrapper = styled.div`
    background: var(--light-green);
    width: 50vw;
    height: 50vw;
    border-radius: 6px;
    display: grid;
    place-items: center;
`

const SpaceNameInput = styled.div`
    border-bottom: 1px solid var(--medium-grey);
    width: 80%;
    margin-bottom: 5px;

    input {
        text-align: center;
        font-weight: 400;
        font-size: 1rem;
        height: 50px;
        margin-top: 5px;
        padding-left: 10px;
        background: var(--light-grey);
        color: var(--dark-grey);
        border: 0px;
    }

    input:focus {
        outline: none;
    }
`
const ButtonWrapper = styled.div`
    display: flex;
    justify-content: center;
    margin-top: 50px;
    width: 80%;
`
const ButtonStyled = styled.button`
    color: var(--white);
    background-color: var(--dark-green);
    border: none;
    border-radius: 5px;
    width: 100%;
    padding: 15px;
`
