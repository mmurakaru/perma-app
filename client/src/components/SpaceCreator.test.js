import SpaceCreator from './SpaceCreator'
import { render } from '@testing-library/react'

describe('SpaceCreator', () => {
    it('renders correctly', () => {
        const { container } = render(<SpaceCreator submitDisabled={true} />)
        expect(container.firstChild).toMatchSnapshot()
    })
})
