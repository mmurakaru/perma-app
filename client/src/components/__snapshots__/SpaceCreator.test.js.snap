// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceCreator renders correctly 1`] = `
.c0 {
  background: var(--lightGrey);
  width: 100%;
  height: 100%;
  padding: 20px;
}

.c3 {
  width: 60%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 15vh;
}

.c1 label {
  font-weight: 200;
  width: 70%;
  text-align: center;
}

.c2 {
  background: var(--lightGreen);
  width: 50vw;
  height: 50vw;
  border-radius: 6px;
  display: grid;
  place-items: center;
}

.c4 {
  border-bottom: 1px solid var(--mediumGrey);
  width: 80%;
  margin-bottom: 5px;
}

.c4 input {
  text-align: center;
  font-weight: 400;
  font-size: 1rem;
  height: 50px;
  margin-top: 5px;
  padding-left: 10px;
  background: var(--lightGrey);
  color: var(--darkGrey);
  border: 0px;
}

.c4 input:focus {
  outline: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 50px;
  width: 80%;
}

.c6 {
  color: var(--white);
  background-color: var(--darkGreen);
  border: none;
  border-radius: 5px;
  width: 100%;
  padding: 15px;
}

<section
  class="c0"
>
  <form
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        class="c3"
      >
        banana.svg
      </svg>
    </div>
    <div
      class="c4"
    >
      <input
        id="spacetitle"
        type="text"
      />
    </div>
    <label
      for="spacetitle"
    >
      What would you like to name your space?
    </label>
    <div
      class="c5"
    >
      <button
        class="c6"
        disabled=""
      >
        Agree and create space
      </button>
    </div>
  </form>
</section>
`;
